jsonFile/jsonFile.go

```diff
@@ -0,0 +1,158 @@
+package jsonFile
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	log "github.com/sirupsen/logrus"
+)
+
+// Aggregator holds the services URLs from Json file.
+type Aggregator struct {
+	services map[string]*NodeInfo
+	r        *resty.Client
+}
+
+// NodeInfo embeds node-related information
+type NodeInfo struct {
+	URL string `json:"url"`
+}
+
+// GetInfoEndpoint returns info endpoint URL
+func (ni *NodeInfo) GetInfoEndpoint() string {
+	infoEndpoint, err := url.JoinPath(ni.URL, "/info")
+	if nil != err {
+		log.Errorf("Unable to join URL: %v", err)
+	}
+
+	return infoEndpoint
+}
+
+// GetHealthEndpoint returns health check URL
+func (ni *NodeInfo) GetHealthEndpoint() string {
+	healthEndpoint, err := url.JoinPath(ni.URL, "/health")
+	if nil != err {
+		log.Errorf("Unable to join URL: %v", err)
+	}
+
+	return healthEndpoint
+}
+
+// NewAggregator creates new aggregator from Json file
+func NewAggregator(configFile string, timeout time.Duration) (*Aggregator, error) {
+	services, err := loadProperties(configFile)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Aggregator{
+		r: resty.NewWithClient(&http.Client{
+			Timeout: timeout,
+		}),
+		services: services,
+	}, nil
+
+}
+
+// LoadServices URLs from a Json file
+func loadProperties(filename string) (map[string]*NodeInfo, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		log.Errorf("Failed to open file: %v", err)
+	}
+	defer file.Close()
+
+	var byteValue []byte
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		byteValue = append(byteValue, scanner.Bytes()...)
+	}
+	if err := scanner.Err(); err != nil {
+		log.Errorf("Failed to read file: %v", err)
+	}
+
+	var nodesInfo map[string]*NodeInfo
+	if err := json.Unmarshal(byteValue, &nodesInfo); err != nil {
```

Comment:
it seems the be we are reading the whole file. In this case it could be as easy as
```
// read our opened jsonFile as a byte array.
byteValue, _ := ioutil.ReadAll(jsonFile)
var nodesInfo map[string]*NodeInfo
json.Unmarshal(byteValue, &nodesInfo)
```